src/pages/account.tsx

```diff
@@ -1,30 +1,181 @@
-import { Box, Button, Typography } from '@mui/material';
-import {getSession, useSession, signOut} from 'next-auth/react';
+import { Avatar, Box, Button, Typography, Dialog, DialogActions, Divider, DialogContent, DialogTitle, TextField } from '@mui/material';
+import { useSession } from 'next-auth/react';
 import type { NextPageWithLayout } from '@common/types';
 import type { ReactElement } from 'react';
 import PageLayout from '@common/layouts/PageLayout';
+import { useState } from 'react';
+import { useRouter } from 'next/navigation';
+
 
 const Account: NextPageWithLayout = () => {
   const { data: session, status } = useSession();
-  console.log(session)
   const user = session?.user;
-  const isLoadingUser = status === 'loading';
+  const [isEditing, setIsEditing] = useState(false);
+  const [username, setUsername] = useState(session?.user?.name ?? '');
+  const [accountType, setAccountType] = useState(session?.user?.type ?? '');
+  const router = useRouter();
+
+  const handleEditClick = () => {
+    setIsEditing(true);
+  };
+
+  const handleSaveClick = async () => {
+    setIsEditing(false);
+    console.log('saving changes to user');
+    // gather the data from the form
+    const formData = { email: email , name: username, type: accountType };
+
+    // send the data to the server TO DO: create API route
+    const response = await fetch('/api/user/updateUser', {
+      method: 'POST',
+      headers: { 'Content-Type': 'application/json' },
+      body: JSON.stringify(formData),
+    });
+    console.log(formData)
+    console.log(response);
+
+    // update the user's session
+    
+  };
+
+  const handleCancelClick = () => {
+    setIsEditing(false);
+  };
 
-  // const isNewUser = user.type === 'new';
+  const name = user?.name || user?.email || 'Rodeo Co User';
+  const email = user?.email;
+  const avatarLetter = name[0].toUpperCase();
+  const avatarSrc = user?.image;
+  const avatarSize = 100;
+
+  let userType;
+  if (user?.type === 'new') {
+    userType = 'New';
+  } else if (user?.type === 'admin') {
+    userType = 'Producer';
+  } else {
+    userType = 'Participant';
+  }
 
-  // todo: if user.type === 'new', make the user choose admin or participant type
-  // ^ this should probably be done in middleware, actually - a forced redirection to /account (this view) until user has a real type
   return (
-    <p>Are you a rodeo administrator looking to create rodeos, or a participant?</p>
-  )
+    <>
+      <Box p={4} sx={{fontFamily:'Poppins, sans-serif'}}>
```

Comment:
Would you be okay with breaking the form into a separate component and loading it into this page? It would make dealing with different user types and sessions easier

Reply:
It would also simplify the merge conflicts I'll need to address, so it's also a selfish request 😂

Reply:
This is done!